### DISCUSSION TITLE
Renaming winmm.dll

### DISCUSSION THREAD

**Discussion by StenApp** [ayuanx/ogg-winmm#9] [Q&A]
Why do i have to rename winmm.dll to e.g. winmx.dll to make it work?

**Comment by YELLO-belly** [comment]
I have heard about this issue before affecting some users. Basically the operating system seems to load only the real system dll file ignoring the local wrapper. Renaming the wrapper and hex editing the executable to point to it is probably the best solution in that case.

But it is a curious thing and I do not know why some users have this issue. One could look at the MS documentation here:
https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order
See the notes on Search Order for Desktop Applications and the list for known dll files in the registry.

For dsound.dll there is talk about a similar issue here:
https://www.indirectsound.com/registryIssues.html
And for ddraw.dll (dgvoodoo2) here:
https://www.vogons.org/viewtopic.php?f=59&t=82073
But these will not work directly with winmm.dll since you need a different CLSID.

**EDIT:**
I also made a small PowerShell script which can automate the renaming of the winmm string in a game executable here:
https://github.com/YELLO-belly/ogg-winmm/tree/master/PS-Script
